cmake/modules/extensions.cmake

```diff
@@ -2507,76 +2538,61 @@ function(zephyr_get variable)
     set(GET_VAR_SYSBUILD "GLOBAL")
   endif()
 
-  if(GET_VAR_MERGE)
-    # Clear variable before appending items in MERGE mode but keep a backup for
-    # local appending later
-    set(local_var_backup ${${variable}})
-    set(${variable})
+  if(NOT DEFINED GET_VAR_VAR)
+    set(GET_VAR_VAR ${variable})
   endif()
 
-  set(used_global false)
-
-  if(SYSBUILD)
-    get_property(sysbuild_name TARGET sysbuild_cache PROPERTY SYSBUILD_NAME)
-    get_property(sysbuild_main_app TARGET sysbuild_cache PROPERTY SYSBUILD_MAIN_APP)
-    get_property(sysbuild_${variable} TARGET sysbuild_cache PROPERTY ${sysbuild_name}_${variable})
-    if(NOT DEFINED sysbuild_${variable} AND
-       ("${GET_VAR_SYSBUILD}" STREQUAL "GLOBAL" OR sysbuild_main_app)
-    )
-      get_property(sysbuild_${variable} TARGET sysbuild_cache PROPERTY ${variable})
-      set(used_global true)
-    endif()
-  endif()
+  # Keep current scope variables in internal variables.
+  # This is needed to properly handle cases where we want to check value against
+  # environment value or when appending with the MERGE operation.
+  foreach(var ${GET_VAR_VAR})
+    set(current_${var} ${${var}})
```

Comment:
I don't see the backup value being restored, only checked against the environmental value

Reply:
it's being restored through this loop:
https://github.com/zephyrproject-rtos/zephyr/blob/53c2430dbe2a4c2ee0a2c536f2ae744ad7dbee5f/cmake/modules/extensions.cmake#L2566-L2567
which goes through here:
https://github.com/zephyrproject-rtos/zephyr/blob/53c2430dbe2a4c2ee0a2c536f2ae744ad7dbee5f/cmake/modules/extensions.cmake#L2487-L2488

The warning itself is a special handling for the environment variable hiding a local one, as such a situation may indicate a user having something unexpected in environment.